battleground_viewer/src/lib.rs

```diff
@@ -191,18 +190,94 @@ impl ConstructViewer {
                 println!("elements: {}", now.elapsed().as_secs_f64());
             }
 
+            let now = std::time::Instant::now();
+
+            /* The rendering steps will look something like this:
+                0) Prerender shadow maps
+                A) Scene render (targets framebuffer)
+                B1) Render depth of non-emissives into depth texture
+                B2) Render emissives into color texture (use B1 as depth texture)
+                C) Write B2 into A additively
+            */
```

Comment:
Yeah, that is indeed a thing that happens. However, without the numbering it becomes harder to relate the steps to the actual code; especially since we are using results from previous steps. Unless we find some way of structuring the render steps more clearly in code, I think we are best of maintaining the synchronization between the top comment and the comments with the steps themselves.

I'm open to any ideas that can make this (admittedly somewhat ad-hoc) section of code more robust.

Reply:
Yeah, it's tricky, we could have renderable `Object`'s register one or multiple `render at` enum's and then keep an enum that describes these steps and render them in that order... :thinking:

Reply:
That's worth an attempt. I'll look into that approach when refactoring `ConstructRender`.